core/os/time.h

```diff
@@ -49,9 +49,14 @@ class Time : public Object {
 	static void _bind_methods();
 	static Time *singleton;
 
+private:
+	const String datetime_error;
```

Comment:
How can it be const if you are assigning stuff to it?

Reply:
I should have removed that, but let's make sure about the thread-safety: can you advise on the best practice to access the macro variable from the outer `get_datetime_error()`?

Reply:
This error variable is the only place where the Time class has state, which is the concern here.

Scenario:

1. Script A uses a Time method.
2. Script B uses a Time method.
3. Script A checks for a validation error, but gets the error from B's call.

I'm not really convinced that this PR is the way to go. Instead, for the use case you mentioned of performing validation without conversion, I would just add a method that performs the validation and returns a String, and expose this method. Then we can use this method in the others in places where the macro is currently used. This way validation can be done separately from conversion, and the Time class stays stateless.

Reply:
I hadn't been thinking about the total state of the class, so you've pointed out the source of the problem.

Was there an original reason why the parse and validation were put inside macros to begin with that needs to be contemplated? It would be easy to do as you suggest - promoting the validation to method.

Reply:
I just remembered the reason, it's because the method would have a ton of arguments, there's a lot of data to pass around. With a macro there is no need to pass anything, the values are already there in the method. If we want to make a validation method I think it's fine to just have a lot of arguments.